### DISCUSSION TITLE
MediaPlayerElement more information

### DISCUSSION THREAD

**Discussion by michalss** [unoplatform/uno#13224] [General]
Hello,

Could you kindly provide information about the MediaPlayerElement?

What is the underlying player framework used (e.g., VLC, MPV, or another)?
Does this implementation have provisions for custom controls?
Is playing from a URL supported?
Does it have the capability to handle multiple audio streams?
Does it offer support for various types of subtitles?
Can I monitor progress, such as the current playback time and player state?
Thank you.

**Comment by michalss** [comment]
found some info here https://platform.uno/docs/articles/controls/MediaPlayerElement.html

but there is nothing about codecs. also need support for audio streams and mostly subtitles, that is very important :) can you please give us some more inside info pls ?

![image](https://github.com/unoplatform/uno/assets/1000729/e7ccf6b9-fa20-4a93-aa38-961855596d3d)

most important things missing :(

**Reply by jeromelaban** [reply]
Thanks for the question. All platforms have different capabilities, so you'll need to be more specific. On iOS, [Android](https://platform.uno/docs/articles/controls/MediaPlayerElement.html#notes) and Wasm, Uno is the native players, and on Skia.Gtk heads, [it's libVLC](https://platform.uno/docs/articles/controls/MediaPlayerElement.html#skiagtk-on-linux).

**Reply by michalss** [reply]
Thank you Jerome for answer. Not sure, im mostly after Android, so is Exoplayer native player for android ? If not, why not implement the best player from gloogle directly and avoid to native one, since Media 3 ExpoPlayer support everything at all out of the box basically :) ? Btw are you also planning to support subtitles and audio tracks please ?

Also can you please share some sample code ?

thank you

**Reply by jeromelaban** [reply]
Currently, we're using this implementation: 
https://github.com/unoplatform/uno/blob/f3490baeadcf28547940ca85edf2a358d36d91bc/src/Uno.UWP/Media/Playback/MediaPlayer.Android.cs#L26

 We have plans to support subtitles and audio tracks, but no specific timeline at this time. If you have specific needs, drop us an email at info@platform.uno!
